Cargo.toml

```diff
@@ -44,6 +44,7 @@ quickcheck-dep = { package = "quickcheck", version = "1.0.3", default-features =
 rand = { version = "0.8.4", optional = true, default-features = false }
 serde = { version = "1.0.126", optional = true, default-features = false }
 time-macros = { version = "=0.2.4", path = "time-macros", optional = true }
+num-traits = { version = "0.2.15", default-features = false }
```

Comment:
Why not extract the one function required instead of adding a dependency? The hardware implementation can still be used, depending on the feature gate.

Reply:
Sure, if you prefer it that way.